src/main/java/games/strategy/engine/data/Unit.java

```diff
@@ -147,11 +147,11 @@ public String toStringNoOwner() {
    * TODO: fix the root cause of this deserialization issue (probably a circular dependency somewhere)
    */
   public static class UnitDeserializationErrorLazyMessage {
-    private static transient boolean s_shownError = false;
+    private static boolean shownError = false;
```

Comment:
`transient` has no effect on `static` fields when the default serializer is used.  Regardless, this nested class is not even `Serializable`.